File: 2810. Faulty Keyboard/Solution.py
from collections import deque


class Solution:
    def finalString(self, s: str) -> str:
        q, flag = deque([]), True
        for c in s:
            if c == "i":
                flag = not flag
            elif flag:
                q.append(c)
            else:
                q.appendleft(c)
        return "".join(q if flag else reversed(q))

File: 2810. Faulty Keyboard/Solution.cpp
#include <algorithm>
#include <string>
using namespace std;

class Solution {
public:
    string finalString(string s)
    {
        int len = 1;
        for (size_t i = 1; i < s.size(); i++) {
            if (s[i] != 'i')
                s[len++] = s[i];
            else
                reverse(s.begin(), s.begin() + len);
        }
        s.resize(len);
        return s;
    }
};
